flink-runtime/src/main/java/org/apache/flink/runtime/util/config/memory/ProcessMemoryUtils.java

```diff
@@ -158,6 +156,25 @@ public JvmMetaspaceAndOverhead deriveJvmMetaspaceAndOverheadFromTotalFlinkMemory
 		return jvmMetaspaceAndOverhead;
 	}
 
+	private MemorySize deriveJvmOverheadFromTotalFlinkMemoryAndOtherComponents(
```

Comment:
I think the method `deriveProcessSpecWithTotalProcessMemory` could also be susceptible to the same problem.

Reply:
there is a similar check in the method `deriveJvmMetaspaceAndOverheadWithTotalProcessMemory`, called before:
`jvmMetaspaceAndOverhead.getTotalJvmMetaspaceAndOverheadSize().getBytes() > totalProcessMemorySize.getBytes()`